src/mds/MDLog.h

```diff
@@ -292,13 +295,13 @@ class MDLog {
   friend class C_OFT_Committed;
 
   void try_to_commit_open_file_table(uint64_t last_seq);
-  LogSegment* _start_new_segment(SegmentBoundary* sb);
+  LogSegmentRef& _start_new_segment(SegmentBoundary* sb);
   void _segment_upkeep();
   LogSegment::seq_t _submit_entry(LogEvent* e, MDSLogContextBase* c);
 
-  void try_expire(LogSegment *ls, int op_prio);
-  void _maybe_expired(LogSegment *ls, int op_prio);
-  void _expired(LogSegment *ls);
+  void try_expire(LogSegmentRef& ls, int op_prio);
+  void _maybe_expired(LogSegmentRef& ls, int op_prio);
+  void _expired(LogSegmentRef& ls);
```

Comment:
I'm not going to add a nit for these but, also, these would be better as `LogSegmentRef const&`.